### PR TITLE
Capturing lavamoat stats in E2E

### PR DESCRIPTION
**Pull request by jpuri**
Capturing initialisation stats in e2e test.

### CODE REVIEW

**Review by darkwing — APPROVED** [review]

File: development/build/scripts.js [hunk 1]
```diff
@@ -370,16 +370,20 @@ const postProcessServiceWorker = (
...
 ) => {
   mv3BrowserPlatforms.forEach((browser) => {
     const appInitFile = `./dist/${browser}/app-init.js`;
     const fileContent = readFileSync('./app/scripts/app-init.js', 'utf8');
-    const fileOutput = fileContent
-      .replace('/** FILE NAMES */', fileList)
-      .replace(
+    let fileOutput = fileContent.replace('/** FILE NAMES */', fileList);
+    if (!testing) {
+      // Lavamoat is always set to true in testing mode
```
Comment (kumavis) [inline development/build/scripts.js:380]: > Lavamoat is always set to true in testing mode

if this is true we shouldnt need to branch on `testing`

File: development/build/scripts.js [hunk 2]
```diff
@@ -420,16 +424,32 @@ async function bundleMV3AppInitialiser({
...
 
-  let prevChromeFileContent;
-  watch('./dist/chrome/app-init.js', () => {
-    const chromeFileContent = readFileSync('./dist/chrome/app-init.js', 'utf8');
-    if (chromeFileContent !== prevChromeFileContent) {
-      prevChromeFileContent = chromeFileContent;
-      postProcessServiceWorker(mv3BrowserPlatforms, fileList, applyLavaMoat);
-    }
-  });
+  if (devMode && !testing) {
```
Comment (kumavis) [inline development/build/scripts.js:434]: could use more comments

File: development/build/scripts.js [hunk 3]
```diff
@@ -420,16 +424,32 @@ async function bundleMV3AppInitialiser({
...
+      );
+      if (chromeFileContent !== prevChromeFileContent) {
+        prevChromeFileContent = chromeFileContent;
+        postProcessServiceWorker(mv3BrowserPlatforms, fileList, applyLavaMoat);
+      }
+    });
+  }
+
+  if (testing) {
+    const content = readFileSync('./dist/chrome/runtime-lavamoat.js', 'utf8');
```
Comment (kumavis) [inline development/build/scripts.js:449]: add comments

**Review by kumavis — APPROVED** [review]
looks nice, small notes

**Comment by jpuri** [comment]
@kumavis : I merged this PR, I will address the feedbacks in following one.

File: development/build/scripts.js (hunk 1 above)
Comment (jpuri) [reply development/build/scripts.js:380]: This is fixed in following PR: https://github.com/MetaMask/metamask-extension/pull/15157/files#diff-a05d1f077953d24c0d8ddecf64163da2653ba3b728e1ba1f18af6d770767b127R379

File: development/build/scripts.js (hunk 3 above)
Comment (jpuri) [reply development/build/scripts.js:449]: Done here: https://github.com/MetaMask/metamask-extension/pull/15157/files#diff-a05d1f077953d24c0d8ddecf64163da2653ba3b728e1ba1f18af6d770767b127R451

File: development/build/scripts.js (hunk 2 above)
Comment (jpuri) [reply development/build/scripts.js:434]: https://github.com/MetaMask/metamask-extension/pull/15157/files#diff-a05d1f077953d24c0d8ddecf64163da2653ba3b728e1ba1f18af6d770767b127R435

File: development/build/scripts.js (hunk 1 above)
Comment (jpuri) [reply development/build/scripts.js:380]: I think this comment is confusing, I changes it here: https://github.com/MetaMask/metamask-extension/pull/15157/files#diff-a05d1f077953d24c0d8ddecf64163da2653ba3b728e1ba1f18af6d770767b127R381

**Pull request merged**